packages/financial-templates-lib/src/price-feed/CreatePriceFeed.ts

```diff
@@ -481,59 +503,69 @@ async function createPriceFeed(logger, web3, networker, getTime, config) {
           .map((node) => node.name)
       )
-    );
+    ).filter(isDefined);
 
     // This is a complicated looking map that maps each symbol into an entry in an object with its value the price
     // feed created from the mapped config in allConfigs.
     const priceFeedMap = Object.fromEntries(
       (
         await Promise.all(
-          symbols.map(async (symbol) => {
-            const config = allConfigs[symbol];
-
-            // If there is no config for this symbol, return just null, which will be filtered out.
-            // Allow this through becuase
-            if (!config) {
-              logger.debug({
-                at: "_createExpressionPriceFeed",
-                message: `No price feed config found for symbol: ${symbol} 🚨`,
-                expressionConfig,
-              });
-              return null;
+          symbols.map(
+            async (symbol: string): Promise<null | [string, PriceFeedInterface | null]> => {
+              const config = allConfigs[symbol];
+
+              // If there is no config for this symbol, return just null, which will be filtered out.
+              // Allow this through becuase
+              if (!config) {
+                logger.debug({
+                  at: "_createExpressionPriceFeed",
+                  message: `No price feed config found for symbol: ${symbol} 🚨`,
+                  expressionConfig,
+                });
+                return null;
+              }
+
+              // These configs will inherit the expression config values (except type), but prefer the individual config's
+              // value when present.
+              const combinedConfig = { ...expressionConfig, type: undefined, ...config };
+
+              // If this returns null, just return upstream since the error has already been logged and the null will be
+              // detected upstream.
+              const priceFeed = await createPriceFeed(logger, web3, networker, getTime, combinedConfig);
+              return [symbol, priceFeed];
             }
-
-            // These configs will inherit the expression config values (except type), but prefer the individual config's
-            // value when present.
-            const combinedConfig = { ...expressionConfig, type: undefined, ...config };
-
-            // If this returns null, just return upstream since the error has already been logged and the null will be
-            // detected upstream.
-            const priceFeed = await createPriceFeed(logger, web3, networker, getTime, combinedConfig);
-            return [symbol, priceFeed];
-          })
+          )
         )
-      ).filter((el) => el !== null)
+      ).filter(isDefined)
     );
 
     // Return null if any of the price feeds in the map are null (meaning there was an error).
-    if (Object.values(priceFeedMap).some((priceFeed) => priceFeed === null)) return null;
+    if (!Object.values(priceFeedMap).every(isDefined)) return null;
```

Comment:
There's a subtlety here. To simplify, the inner most function is returning `[string, PriceFeed | null] | null`. The first filter drops the outer null, the second check is checking for the inner null. The distinction is important because a failure to find a price feed for a symbol in the expression isn't necessarily an error (the symbol could be a built-in or resolved in some other way). If we try to generate an inner price feed and it fails and then returns null, we want to propagate that error by continuing to return null up the stack. 

In hindsight, we should probably throw instead of returning nulls, but that's the pattern we used 🤷 .

Reply:
yea thanks for the explanation